packages/frontend/src/components/ChartConfigPanel/Legend/ReferenceLines.tsx

```diff
@@ -0,0 +1,154 @@
+    TableCalculation,
+} from '@lightdash/common';
+import { FC, useEffect, useMemo, useState } from 'react';
+import FieldAutoComplete from '../../common/Filters/FieldAutoComplete';
+import { useVisualizationContext } from '../../LightdashVisualization/VisualizationProvider';
+import { GridSettings, SectionTitle } from './Legend.styles';
+
+type Props = {
+    items: (Field | TableCalculation | CompiledDimension)[];
+};
+
+export const ReferenceLines: FC<Props> = ({ items }) => {
+    const {
+        cartesianConfig: { dirtyLayout, dirtyEchartsConfig, updateSeries },
+    } = useVisualizationContext();
+
+    const fieldsInAxes = useMemo(() => {
+        const fieldNames = [
+            dirtyLayout?.xField,
+            ...(dirtyLayout?.yField || []),
+        ];
+        return items.filter((item) => {
+            const fieldId = isField(item) ? getFieldId(item) : item.name;
+            // Filter numeric fields (remove if we start supporting other types)
+            return fieldNames.includes(fieldId) && isNumericItem(item);
+        });
+    }, [items, dirtyLayout]);
+
+    const [selectedMarklineAxis, selectedMarklineValue, selectedFieldId] =
+        useMemo(() => {
+            const serieWithMarkLine = dirtyEchartsConfig?.series?.find(
+                (serie) => serie.markLine?.data[0] !== undefined,
+            );
+
+            const markLine = serieWithMarkLine?.markLine?.data[0];
+            if (markLine === undefined)
+                return [undefined, undefined, undefined];
+            const [markLineKey, markLineValue] = Object.entries(markLine)[0];
+            const fieldId =
+                markLineKey === 'xAxis'
+                    ? serieWithMarkLine?.encode.xRef.field
+                    : serieWithMarkLine?.encode.yRef.field;
+            return [markLineKey, markLineValue, fieldId];
+        }, [dirtyEchartsConfig?.series]);
+
+    const [value, setValue] = useState<string | undefined>(
+        selectedMarklineValue,
+    );
+
+    const selectedFieldDefault = useMemo(() => {
+        if (selectedMarklineAxis === undefined) return;
+        return fieldsInAxes.find((field) => {
+            const fieldId = isField(field) ? getFieldId(field) : field.name;
+            return fieldId === selectedFieldId;
+        });
+    }, [fieldsInAxes]);
+
+    const [selectedField, setSelectedField] = useState<
+        Field | TableCalculation | CompiledDimension | undefined
+    >(selectedFieldDefault);
+
+    const [isOpen, setIsOpen] = useState<boolean>(
+        selectedMarklineAxis !== undefined,
+    );
+
+    useEffect(() => {
+        if (value && selectedField) {
+            const fieldId = isField(selectedField)
+                ? getFieldId(selectedField)
+                : selectedField.name;
+
+            if (dirtyEchartsConfig?.series) {
+                let alreadyAdded = false; // flag to avoid duplication on Xaxis
+                const series = dirtyEchartsConfig?.series.map((serie) => {
```

Comment:
Once we have multiple reference lines, we can replace this `reset` and updateAll with a updateSingleSeries